Add build-time option for OMP scheduler; document MULTITHREAD_THRESHOLD range

martin-frbg opened the pull request:
to address topics mentioned in #103

martin-frbg committed:
Allow changing the OpenMP scheduler

martin-frbg committed:
Allow choosing the OpenMP scheduler and add range hint for GEMM_MULTITHREAD_THRESHOLD

martin-frbg committed:
Leave DEBUG unset by default

sandwichmaker commented on Makefile.rule:
```diff
@@ -157,7 +165,8 @@ NO_AFFINITY = 1
 
 # If any gemm arguement m, n or k is less or equal this threshold, gemm will be execute
 # with single thread. You can use this flag to avoid the overhead of multi-threading
```
This documentation is not quite correct, since this threshold is now used a multiplier to a per function constant and has nothing to do directly with particular matrix dimensions.

martin-frbg replied:
That's right, though as far as I tried to understand the transition the idea was that it was closer to the originally intended effect. Not sure how to phrase this, unless we want to call it a controversial magic number with dubious real-world equivalent :grinning:

martin-frbg committed:
Amended description of GEMM_MULTITHREAD_THRESHOLD

to reflect #742 making it track floating point operations rather than matrix size
